pilot/pkg/serviceregistry/kube/controller/endpoint_builder.go

```diff
@@ -32,33 +33,39 @@ type EndpointBuilder struct {
 	controller *Controller
 
 	labels         labels.Instance
-	uid            string
 	metaNetwork    string
 	serviceAccount string
 	locality       model.Locality
 	tlsMode        string
+	workloadName   string
+	namespace      string
 }
 
 func NewEndpointBuilder(c *Controller, pod *v1.Pod) *EndpointBuilder {
-	locality, sa, uid := "", "", ""
+	locality, sa, wn, namespace := "", "", "", ""
 	var podLabels labels.Instance
 	if pod != nil {
 		locality = c.getPodLocality(pod)
 		sa = kube.SecureNamingSAN(pod)
-		uid = createUID(pod.Name, pod.Namespace)
 		podLabels = pod.Labels
+		namespace = pod.Namespace
+	}
+	dm, _ := kubeUtil.GetDeployMetaFromPod(pod)
```

Comment:
why not use pod.Name directly

Reply:
What we need is workload name, which will be used for metric label.

Reply:
IC